src/main/java/com/google/devtools/build/lib/runtime/commands/InfoCommand.java

```diff
@@ -213,37 +213,27 @@ public BlazeCommandResult exec(
                     "Blaze info does not support starlark options. Ignoring options: "
                         + removedStarlarkOptions));
       }
-      if (residue.size() > 1) {
-        String message = "at most one key may be specified";
-        env.getReporter().handle(Event.error(message));
-        return createFailureResult(
-            message, ExitCode.COMMAND_LINE_ERROR, FailureDetails.InfoCommand.Code.TOO_MANY_KEYS);
-      }
 
-      String key = residue.size() == 1 ? residue.get(0) : null;
       env.getEventBus().post(new NoBuildEvent());
-      if (key != null) { // print just the value for the specified key:
-        byte[] value;
-        if (items.containsKey(key)) {
-          value = items.get(key).get(configurationSupplier, env);
-        } else {
-          String message = "unknown key: '" + key + "'";
-          env.getReporter().handle(Event.error(message));
-          return createFailureResult(
-              message,
-              ExitCode.COMMAND_LINE_ERROR,
-              FailureDetails.InfoCommand.Code.KEY_NOT_RECOGNIZED);
-        }
-        try {
-          outErr.getOutputStream().write(value);
-          outErr.getOutputStream().flush();
-        } catch (IOException e) {
-          String message = "Cannot write info block: " + e.getMessage();
-          env.getReporter().handle(Event.error(message));
-          return createFailureResult(
-              message,
-              ExitCode.ANALYSIS_FAILURE,
-              FailureDetails.InfoCommand.Code.INFO_BLOCK_WRITE_FAILURE);
+      if (residue.size() > 0) {
+        for (String key : residue) {
+          byte[] value;
+          if (items.containsKey(key)) {
+            try (SilentCloseable c = Profiler.instance().profile(key + ".infoItem")) {
+              value = items.get(key).get(configurationSupplier, env);
+              if (residue.size() > 1) {
+                outErr.getOutputStream().write((key + ": ").getBytes(StandardCharsets.UTF_8));
+              }
+              outErr.getOutputStream().write(value);
+            }
+          } else {
+            String message = "unknown key: '" + key + "'";
```

Comment:
Is it intentional that you error out on the first unknown key? This may not be easy to change when people have started relying on it. Maybe first check all key names for validity before printing?

Reply:
Done in a slightly different way: I collect all unknown keys and report them in the end.